lib/sles4sap_publiccloud.pm

```diff
@@ -1014,26 +1015,29 @@ sub saphanasr_showAttr_version {
 
 =item B<TIMEOUT> - The number of seconds to wait before aborting zypper ref
 
+=item B<runas> - If 'runas' defined, command will be executed as specified user, otherwise it will be executed as cloudadmin.
+
 =back
 =cut
 
 sub wait_for_zypper {
     my ($self, %args) = @_;
     croak("Argument <instance> missing") unless ($args{instance});
-    $args{max_retries} //= 10;
-    $args{retry_delay} //= 20;
-    $args{timeout} //= 600;
-    my $retries = 0;
-
-    while ($retries < $args{max_retries}) {
-        my $ret = $args{instance}->run_ssh_command(cmd => 'sudo zypper ref', username => 'cloudadmin', proceed_on_failure => 1, rc_only => 1, quiet => 1, timeout => $args{timeout});
+    my $max_retries = $args{max_retries} // 10;
```

Comment:
Why change from `$args{max_retries} //= 10;` to `my $max_retries = $args{max_retries} // 10;` is needed ? Does it has impact on ssh stability or is more about code style? If it is the second I personally usually suggest the opposite, to prefer `$args{max_retries} //= 10;` . My personal reason behind it is to avoid multiple variables with same name when not strictly needed.

`my $max_retries = $args{max_retries} // 10;` is needed when you need both values:
1. a local variable that can be used to do something 
2. a "backup" of the original value of the input argument

Reply:
I think it's even more efficient to use `$args{named_variable}` instead of `$named_variable` because that way you're not duplicating things in memory. Only advantage of using `$named_variable` is with regards to code legibility (fewer `{` and `}` everywhere around the code), but other than that I also vote to stick with `$args{named_variable}`.

Then again, I submitted [this](https://github.com/os-autoinst/openQA/pull/1559/files#diff-e95ca4431cfcef5e959084364aacfa66b73b09181338837c249f316481f6d12a) back in the day so I may be biased.

Reply:
Just because the rest of the lib use it this way and I don't want to change it whole. :-)